api/addjob.go

```diff
@@ -0,0 +1,79 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/ONSdigital/dp-import-api/models"
+	"github.com/ONSdigital/go-ns/audit"
+	"github.com/ONSdigital/go-ns/common"
+	"github.com/ONSdigital/go-ns/log"
+	"github.com/pkg/errors"
+)
+
+func (api *ImportAPI) addJobHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	// record attempt to add a job
+	if auditError := api.auditor.Record(ctx, addJobAction, audit.Attempted, nil); auditError != nil {
+		handleErr(ctx, w, auditError, nil)
+		return
+	}
+
+	// marshal request body into job structure
+	job, err := models.CreateJob(r.Body)
+	defer r.Body.Close()
+	if err != nil {
+		log.ErrorCtx(ctx, errors.WithMessage(err, "api endpoint addJob error - Bad client request received"), nil)
+
+		// record failure to add job
+		if auditError := api.auditor.Record(ctx, addJobAction, audit.Unsuccessful, nil); auditError != nil {
+			err = auditError
+		}
+
+		handleErr(ctx, w, err, nil)
+		return
+	}
+
+	logData := log.Data{"recipeID": job.RecipeID}
+	auditParams := common.Params{"recipeID": job.RecipeID}
+
+	b, err := api.addJob(ctx, job, auditParams, logData)
+	if err != nil {
+		// record unsuccessful attempt to add job
+		if auditError := api.auditor.Record(ctx, addJobAction, audit.Unsuccessful, auditParams); auditError != nil {
+			err = auditError
+		}
+
+		handleErr(ctx, w, err, logData)
+		return
+	}
+	// record successful attempt to add job
+	api.auditor.Record(ctx, addJobAction, audit.Successful, auditParams)
+
+	w.Header().Set("Content-Type", "application/json")
```

Comment:
You're doing this inside the writeBody function as well

Reply:
Header `Content-Type` needs to be set before calling function WriteHeader() otherwise it cannot write the header.

Other handlers which call `writeBody()` will not have set the header `Content-Type` and so it exists in the `writeBody()` function also.

Lastly the reason why i've left the content-type being set in `writeBody()` is so I don't duplicate the code elsewhere.

Reply:
So after discussion with @mattrout92 , we agreed that the `writeBody()` func should be updated to be generic (change name) and take in a status code, so it can explicitly set the response header and body.